---
title: 'Assignment 2: Differential Gene expression and Preliminary ORA'
author: "Advait Sridhar"
date: "`r Sys.Date()`"
output:
  html_document:
    df_print: paged
    toc: true
    theme: united
    toc_depth: 3
---

\tableofcontents


# Brief Introduction 
The objective of this report is to perform differential gene expression analysis and preliminary thresholded over-representation analysis on a chosen Bulk RNA-seq dataset, which can be found here: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE173940. The initial processing was done in the first assignment, where the count file was downloaded through Bioconductor using GEO id GSE173940. Data was filtered by only keeping rows with significant counts. The rows had ensembl gene ids, which were mapped to HUGO gene symbols using biomaRt. Data was cleaned as the rows which contained pseudogenes or no conversion to HUGO symbols were removed, and rows with duplicate ensembl gene ids were removed as well. Data was normalized by calculating normalization factors using edgeR package and applying these to the dataset. Normalization was checked using boxplot, density plot, MDS plot, BCV plot, and mean-var plot. These filtered, normalized, and cleaned data was downloaded in a txt file which can be found in "A2_data". This data will be used for this report to calculate differential expression in various samples as well as conduct preliminary ORA. 

```{r}
#Setting Working Directory
setwd("/home/rstudio/projects")
getwd()
```

```{r}
#Installing and Loading Required Packages
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
if (!requireNamespace("knitr", quietly = TRUE))
    install.packages("knitr")
if (!requireNamespace("Biobase", quietly = TRUE))
    BiocManager::install("Biobase")
if (!requireNamespace("limma", quietly = TRUE))
    install.packages("limma")
if (!requireNamespace("ComplexHeatmap", quietly = TRUE))
    BiocManager::install("ComplexHeatmap")
if (!requireNamespace("circlize", quietly = TRUE))
    install.packages("circlize")
if (!requireNamespace("gprofiler2", quietly = TRUE))
    install.packages("gprofiler2")

library(BiocManager)
library(knitr)
library(Biobase)
library(limma)
library(ComplexHeatmap)
library(circlize)
library(gprofiler2)
```

Downloading the Data

```{r}
#Reading filtered normalized cleaned data from txt file
atm_count_data <- read.table(file=file.path(getwd(), "A2_data", 
                              "atm_exp_filtered_norm_clean.txt"),
                                    header = TRUE,sep = "\t",
                                    stringsAsFactors = FALSE,
                                    check.names=FALSE)

kable(atm_count_data[1:5,1:5], type="html")
```

Creating Samples Dataframe

```{r}
#creating list of treatments that weren't included in original data
treatment <- c("WT", "Camptothecin", "DMSO", "Camptothecin", "DMSO", "Camptothecin", "DMSO", "Camptothecin", "ATM-KO", "WT", "ATM-KO", "WT", "ATM-KO", "WT", "ATM-KO", "DMSO", "ATM-KO", "ATM-KO", "ATM-KO", "ATM-KO", "DMSO", "DMSO", "DMSO", "DMSO", "Camptothecin", "Camptothecin", "Camptothecin", "Camptothecin", "DMSO", "DMSO", "DMSO", "DMSO")

#creating list of cell lines used for experimentation
cell_line <- c("U2OS", "HEK293T", "HEK293T", "HEK293T", "HEK293T", "HEK293T", "HEK293T", "HEK293T", "U2OS", "U2OS", "U2OS", "U2OS", "U2OS", "U2OS", "U2OS", "HEK293T", "HEK293T", "HEK293T", "HEK293T", "HEK293T", "HEK293T", "HEK293T", "HEK293T", "HEK293T", "U2OS", "U2OS", "U2OS", "U2OS", "U2OS", "U2OS", "U2OS", "U2OS")

#compiling cell lines and treatment information into a data frame
samples <- data.frame(cell_line, treatment)

#changing rownames to sample names in count data
rownames(samples) <- colnames(atm_count_data[,3:ncol(atm_count_data)])

#creating new sample names with information of patient number, cell line, and treatment group
samples$sample_name <- paste(as.numeric(gsub("Wouter", "", rownames(samples))), samples$cell_line, samples$treatment, sep = "_")

samples[1:5,]
```

# Differential Gene Expression

The aim of this report is to perform differential gene expression analysis using the normalized counts file. We can do this by firstly creating an expression matrix dataset containing count data prior to analysis within their respective samples. (Isserlin, 2022)

```{r}
#Creating an expression matrix to analyze differential expression between genes
expression_matrix <- as.matrix(atm_count_data[,3:34])

#Changing rownames to ensembl ids for accurate mapping
rownames(expression_matrix) <- atm_count_data$ensembl_id

#Changing column names to combined sample names
colnames(expression_matrix) <- colnames(atm_count_data)[3:34]

#Creating minimal set using expression matrix
minimal_set <- ExpressionSet(assayData = expression_matrix)
```

Initial model design only takes into account the treatment for each sample, of which there are 4 treatments for 32 samples. The matrix model below reflects this.

```{r}
atm_model_design <- model.matrix(~ samples$treatment )

kable(atm_model_design[1:5,], type="html")
```

Using this initial model, we can create a linear model using the treatments with their respective samples and counts to create a MArrayLM object. Empirical Bayes was applied to this object to get preliminary stats on differential expression for each gene in the cleaned dataset. The gene names were combined with these statistics and the output can be viewed below. (Isserlin, 2022)

```{r}
#Linear Model using minimal set and model with cell lines and treatments
atm_fit <- lmFit(minimal_set, atm_model_design)

#Empirical Bayes for computing stats of linear model 
atm_fit2 <- eBayes(atm_fit,trend=TRUE)

#Creating table with stats of differential expression for each gene and associated p values
atm_topfit <- topTable(atm_fit2, coef=ncol(atm_model_design), adjust.method = "none", number = nrow(expression_matrix))

#Merging HUGO gene symbols with data
atm_output_hits <- merge(atm_count_data[,1:2], atm_topfit,by.y=0,by.x=1,all.y=TRUE)

#Sorting by p value in ascending order
atm_output_hits <- atm_output_hits[order(atm_output_hits$P.Value),]

kable(atm_output_hits[1:10,1:8],type="html",row.names = FALSE)
```

The number of genes with significant differential expression was calculated and ended up to be 1364 prior to correction. Threshold was set at p < 0.05 as this is a standard for differential expression data and was used by the researchers in their paper, albeit arbitrarily.

```{r}
#Number of genes with significant differential expression
length(which(atm_output_hits$P.Value < 0.05))
```


A corrected model matrix was created using interactions between both the 2 cell lines and the 4 treatments that were being investigated by the researchers. 

```{r}
#Designing model using matrix of sample cell lines and respective treatments
atm_model_design_corrected <- model.matrix( ~ samples$cell_line + samples$treatment)

kable(atm_model_design_corrected[1:5,1:5],type="html")
```

Another linear model was used to analyze the relatedness of the gene counts to the corrected model. Empirical Bayes was applied to this object to get preliminary stats on differential expression for each gene in the cleaned dataset. Multiple hypothesis testing was done using the bonferroni correction, which was used by the researchers in the original paper to perform all possible pairs of comparisons (Huiting et al, 2021). The gene names were combined with these statistics and the output can be viewed below.

```{r}
#Linear Model using minimal set and model with cell lines and treatments
atm_fit_corrected <- lmFit(minimal_set, atm_model_design_corrected)

#Empirical Bayes for computing stats of linear model 
atm_fit2_corrected <- eBayes(atm_fit_corrected,trend=TRUE)

#Creating table with stats of differential expression for each gene and associated p values
atm_topfit_corrected <- topTable(atm_fit2_corrected, coef=ncol(atm_model_design_corrected), adjust.method = "bonferroni", number = nrow(expression_matrix))

#Merging HUGO gene symbols with data
atm_output_hits_corrected <- merge(atm_count_data[,1:2], atm_topfit_corrected,by.y=0,by.x=1,all.y=TRUE)

#Sorting by p value in ascending order
atm_output_hits_corrected <- atm_output_hits_corrected[order(atm_output_hits_corrected$P.Value),]

kable(atm_output_hits_corrected[1:10,1:8],type="html",row.names = FALSE)
```

The number of genes after model correction which were significantly expressed was 2017 using threshold p < 0.05. 

```{r}
#Number of genes with significant differential expression
length(which(atm_output_hits_corrected$P.Value < 0.05))
```

The number of corrected genes were greater than those that were significantly expressed prior to correction, so the genes of interest lie in their commonly significantly expressed genes. This was investigated through plotting genes based on their p-value in both models, and finding the intersecting region that would lead to significantly expressed genes that we are interested in. The list of genes in each model and their associated p values were combined into a dataframe, and these 2 dataframes were merged to investigate each gene based on its p-value in both models. Colors were used to express the various groups in the plot: the genes that had no significant expression in either model were black, genes with significant expression only in the initial model were orange, genes with significant expression in the corrected model were blue, and genes with significant expression in both models were red. (Isserlin, 2022)

```{r}
#Creating dataframes with gene ids and p values from 2 models
atm_model_pvalues <- data.frame(ensembl_id = atm_output_hits$hgnc_symbol, atm_pvalue=atm_output_hits$P.Value)
atm_model_pvalues_corrected <-  data.frame(ensembl_id = atm_output_hits_corrected$hgnc_symbol, 
                                           atm_pvalue_corrected = atm_output_hits_corrected$P.Value)

#Merging values from both models and assigning color based on significant p values
two_models_pvalues <- merge(atm_model_pvalues, atm_model_pvalues_corrected,by.x=1,by.y=1)
two_models_pvalues$colour <- "black"
two_models_pvalues$colour[two_models_pvalues$atm_pvalue<0.05] <- "orange"
two_models_pvalues$colour[two_models_pvalues$atm_pvalue_corrected<0.05] <- "blue"
two_models_pvalues$colour[two_models_pvalues$atm_pvalue<0.05 & two_models_pvalues$atm_pvalue_corrected<0.05] <- "red"

#Plotting both models p values against each other
plot(two_models_pvalues$atm_pvalue,
     two_models_pvalues$atm_pvalue_corrected,
     col = two_models_pvalues$colour,
     xlab = "Simple model p-values",
     ylab ="Corrected model p-values", 
     main="Simple vs Corrected Model p-values")

legend(-0.03,1.03,legend=c("Simple Genes of Interest","Corrected Genes of Interest", "Shared Genes of Interest", "Rest"),
       fill=c("orange","blue", "red", "black"),cex = 0.7)
```

The merged dataframe was used to discern which genes were significantly expressed in both models, which are our genes of interest. This list of genes were inputted into a variable atm_top_hits, which was then used for visualization in a Volcano plot. Following the principle in the previous plot, 2 dataframes containing the logFC values and the negative log of the p values were merged with the gene names. The corrected model was used for this as it contained more genes that were differentially expressed, and so our genes of interest would be highlighted with greater contrast. The merged dataframe were queried with the list of genes of interest, which were plotted in red against the insignificantly expressed genes which were taken from the initial data. Data was highly spread out, so x-axis scale was set to (-50,50) for clearer visualization.

```{r}
#Creating list of genes of interest
atm_top_hits <- two_models_pvalues$ensembl_id[two_models_pvalues$colour == "red"]

#Creating 2 dataframes with logFC and -log(pvalue) from corrected output hits
atm_model_logFC <- data.frame(ensembl_id = atm_output_hits_corrected$hgnc_symbol, atm_model_logFC=atm_output_hits_corrected$logFC)
atm_model_nlog_pvalues <-  data.frame(ensembl_id = atm_output_hits_corrected$hgnc_symbol, 
                                           atm_model_nlog_pvalues = -log10(atm_output_hits_corrected$P.Value))

#Merging dataframes and assigning color based on presence of genes of interest
two_models_volcano <- merge(atm_model_logFC, atm_model_nlog_pvalues,by.x=1,by.y=1)
two_models_volcano$colour <- "black"
two_models_volcano$colour[two_models_volcano$ensembl_id %in% atm_top_hits] <- "red"

#Creating Volcano plot
plot(two_models_volcano$atm_model_logFC,
     two_models_volcano$atm_model_nlog_pvalues,
     col = two_models_volcano$colour,
     xlab = "logFC",
     ylab ="-log10(p-value)",
     xlim = c(-50,50),
     main="logFC vs -log10(p-value) Genes")

legend(-53,4.5, legend=c("Genes of Interest", "Rest"), fill=c("red", "black"), cex = 0.7)

```

The number of genes that were differentially expressed in both models were combined into a variable with size 1234. These are the number of significantly differentially expressed genes after correction and plotting, which is lower than found in either model and constitute the genes of interest for further analysis.
```{r}
#Number of significantly differentially expressed genes from both models
length(atm_top_hits)
```

The top hits from our models were used to generate a heatmap to analyze how differential gene expression varied across samples. An initial heatmap matrix was created to include only the count data, with rownames being the specific ensembl gene ids and the column names being the detailed sample name with information about the patient number, cell line and treatment.  

```{r}
#creating heatmap matrix with count data excluding gene ids
atm_heatmap_matrix <- atm_count_data[,3:ncol(atm_count_data)]

#renaming rows to ensembl ids
rownames(atm_heatmap_matrix) <- atm_count_data$ensembl_id

#using combined sample names for each column
colnames(atm_heatmap_matrix) <- samples$sample_name
```

The heatmap matrix was scaled to show counts as relative to one another, with the median expression value being zero and all above it having a positive value and those below it having a negative value indicating upregulated and downregulated genes. This matrix was scaled again to include only the genes of interest as seen in our atm_top_hits object, and calculate relative expression values for these significantly differentially expressed genes. grep() function was used to arrange the samples by their treatment group, which the last term in each sample name represented. Finally, an if-else statement was used to show upregulated genes as increasingly red in color and downregulated genes as increasingly blue, while the rest remained white. The final heatmap was plotted using the Heatmap() function, which can be seen below. 

```{r}
#Creating scaled matrix with count data
atm_heatmap_matrix_scaled <- t(scale(t(atm_heatmap_matrix)))

#Querying matrix with list of genes of interest
atm_heatmap_matrix_tophits <- t(scale(t(atm_heatmap_matrix_scaled[which((two_models_pvalues$ensembl_id) %in% atm_top_hits),])))

#Ordering matrix by treatment group
atm_heatmap_matrix_tophits<- atm_heatmap_matrix_tophits[ ,
      c(grep(colnames(atm_heatmap_matrix_tophits),
                          pattern = "\\_WT"),
      grep(colnames(atm_heatmap_matrix_tophits),
                          pattern = "\\ATM-KO"),
       grep(colnames(atm_heatmap_matrix_tophits),
                          pattern = "\\Camptothecin"),
       grep(colnames(atm_heatmap_matrix_tophits),
                          pattern = "\\DMSO"))]

#Assigning colors based on relative expression values
if(min(atm_heatmap_matrix_tophits) == 0){
    atm_heatmap_col <- colorRamp2(c( 0, max(atm_heatmap_matrix_tophits)), 
                             c( "white", "red"))
  } else {
    atm_heatmap_col <- colorRamp2(c(min(atm_heatmap_matrix_tophits), 0,
                    max(atm_heatmap_matrix_tophits)), 
                    c("blue", "white", "red"))
  }

#Creating heatmap
atm_heatmap <- Heatmap(as.matrix(atm_heatmap_matrix_tophits),
                           cluster_rows = TRUE,
                           cluster_columns = FALSE,
                               show_row_dend = TRUE,
                               show_column_dend = FALSE, 
                               col=atm_heatmap_col,
                               show_column_names = TRUE, 
                               show_row_names = FALSE,
                               show_heatmap_legend = TRUE,
                               column_names_gp = gpar(fontsize = 6),
                               heatmap_legend_param = list(title = "Expression"),
                               name = "Heatmap"
                               )

atm_heatmap
```

In this heatmap, the conditions appear to mostly cluster although there are some exceptions. For example, U20S cells from patient 1 had a WT treatment but the genes appear to be significantly downregulated compared to other conditions with the same cell line and treatment. The same can be said for HEK293T cells taken from patient 3 that had undergone the DMSO treatment. U20S cells from patient 16 that had undergone the ATM knockout treatment, which is our main treatment of interest, appear to be significantly up-regulated compared to its counterparts. In the case of the ATM-KO cell, the ATM kinase could have been an essential suppressor of the transcription of certain genes, and its knockout could have yielded strong expression while this was not the case for the other KO cells. DMSO could have disturbed transcription in that particular HEK293T cell, leading to significantly downregulated gene expression (Huiting et al, 2021). This will be investigated further in the interpretation section by drawing from conclusions in the original paper for this dataset.

# Thresholded over-representation analysis

After conducting some tests to analyze differential expression for our given set of genes, we need to query our up- and down- regulated genes against some annotation databases to discern the nature of the genes being selected for across samples in this experiment. First, this involved generating a corrected top hits data frame which included eBayes statistics for our list of genes of interest. A new column for rank, which is the approximate value obtained from the volcano plot, was included and the dataframe was ordered by the value of this column.  Lists of upregulated and downregulated genes were created by selecting for genes that had positive logFC values (upregulated) or negative logFC values (downregulated) since all the genes in the list already have a p-value less than 0.05. The lists for upregulated, downregulated, and ranked sets of genes were all written into txt files which can be found in A2_data.

```{r}
#Merging dataframe with HUGO gene symbols in list of genes of interest
atm_top_hits_corrected <- atm_output_hits_corrected[atm_output_hits_corrected$hgnc_symbol %in% atm_top_hits,]

#Ordering by rank based on -log(pvalue) and logFC value
atm_top_hits_corrected[,"rank"] <- -log(atm_top_hits_corrected$P.Value, base = 10) * sign(atm_top_hits_corrected$logFC)

atm_top_hits_corrected <- atm_top_hits_corrected[order(atm_top_hits_corrected$rank),]

atm_ranked_genes <- data.frame(genename = atm_top_hits_corrected$hgnc_symbol, F_stat = atm_top_hits_corrected$rank)
atm_ranked_genes <- atm_ranked_genes[order(atm_ranked_genes$F_stat, decreasing = TRUE),]

#Creating lists of upregulated and downregulated genes based on logFC values (p < 0.05)                                        
atm_upregulated_genes <- atm_top_hits_corrected$hgnc_symbol[which(atm_top_hits_corrected$logFC > 0)]

atm_downregulated_genes <- atm_top_hits_corrected$hgnc_symbol[which(atm_top_hits_corrected$logFC < 0)]

#Writing txt files for lists of upregulated, downregulated, and ranked genes
write.table(atm_upregulated_genes,
            file=file.path("A2_data","atm_upregulated_genes.txt"),sep = "\t",
            row.names = FALSE,col.names = FALSE,quote = FALSE)

write.table(atm_downregulated_genes,
            file=file.path("A2_data","atm_downregulated_genes.txt"),sep = "\t",
            row.names = FALSE,col.names = FALSE,quote = FALSE)

write.table(atm_ranked_genes,
            file=file.path("A2_data","atm_ranked_genes.txt"),sep = "\t",
            row.names = FALSE,col.names = FALSE,quote = FALSE)
```

Geneset enrichment analysis was performed using the gprofiler2 package, which queries g:profiler and associated annotation databases to see how many hits are returned for a queried gene (Kolberg et al, 2020). This method uses the fisher's exact test, which is a gene list enrichment analysis strategy that is useful for our case since the ranked gene list was not used for this part of the analysis and we just used a threshold of p < 0.05 (Isserlin, 2022). Genes were queried against the "hsapiens" annotation database with gprofiler version e105_eg52_p16_e84549f. Information about the relevant annotation databases can be found below. 

```{r}
get_version_info(organism = "hsapiens")
```


Gene list enrichment analysis was done for all the top hits together, the upregulated genes, and then the downregulated genes. A table with the queried genes and their respective term names was generated, as well as a plot to show the hits with their associated annotation databases on a background of all annotated genes. Custom background with genes only from our experiment was not used due to poor output for downregulated genes. The results from each query and their respective outputs can be seen below.

```{r}
atm_gost <- gost(query = atm_top_hits, organism = "hsapiens")
atm_gost$result$query <- "Top Hits from Query of Genes of Interest"
head(atm_gost$result)
```

```{r}
gostplot(atm_gost, capped = FALSE, interactive = FALSE)
```


```{r}
atm_gost_up <- gost(query = atm_upregulated_genes, organism = "hsapiens")
atm_gost_up$result$query <- "Top Hits from Query of Upregulated Genes"
head(atm_gost_up$result)
```

```{r}
gostplot(atm_gost_up, capped = FALSE, interactive = FALSE)
```


```{r}
atm_gost_down <- gost(query = atm_downregulated_genes, organism = "hsapiens")
atm_gost_down$result$query <- "Top Hits from Query of Downregulated Genes"
head(atm_gost_down$result)
```

```{r}
gostplot(atm_gost_down, capped = FALSE, interactive = FALSE)
```


For each query, one gene list was returned with terms from 11 databases. The GLEA performed for the entire list of top hits generated a plot that was the upregulated and downregulated genes put together. Plotting the upregulated and downregulated sets of genes separately simply parsed the whole set of genes into one that had positive logFC values and one that had negative logFC values. They all maintained the same background of all annotated genes. Most of the results were in the GO:BP annotation database for both sets of genes and TF for upregulated genes.


# Interpretation

The top hit term names for the upregulated genes was 'Ribosome, cytoplasmic', indicating that the ATM-KO experiment and its associated treatments had a large impact on ribosomal genes and the ATM kinase perhaps played a role in its suppression. The top 6 hits all had to do with ribosomal proteins or RNA, supporting this claim. The researchers results suggest that topoisomerase poisoning through DMSO and Camptothecin and ATM impairement result in widespread aggregation of LLPS (liquid-liquid phase separation) prone and supersaturated proteins (Huiting et al, 2021). Although the consensus isn't heavy, there have been studies showing ribosomal proteins that are LLPS-prone such as NPM1 (Mitrea, 2018). The researchers didn't have much focus on ribosomal proteins in their paper, meaning that either they disregarded these results or there was a difference in measurement techniques that I didn't take into account in this report. 

The top hits for downregulated genes mostly had to do with developmental processes, but the very top term was the ADRB1-CXCR4 complex, which is involved in the regulation of blood pressure. Although there aren't specific mentions in the paper made about these processes, topoisomerase enzymes which were a focus of the paper are highly influential on gene regulation for developmental processes (Li et al, 2016). Interruptions of these processes can lead to genotoxicity, stunted growth, and even cancer, which was a point the researchers made in their paper (Huiting et al, 2021). 

The results from the plots, heatmap, and enrichment analysis determined a list of significantly differentially expressed genes that were related to the researchers conclusions, albeit indirectly. There are key insights that can be gained from understanding the functions of the genes that displayed significant differential expression between samples. Samples that had Camptothecin or DMSO as treatments were more likely to experience topoisomerase degradation due to the ability of these treatments to cause relaxation of coiling (Huiting et al, 2021). This may have caused relatively downregulated gene expression, as is seen especially in the case of HEK293T cells treated with Camptothecin. This is expected to occur with ATM-KO samples as well since ATM kinase is involved in DNA repair and its knockout could lead to downregulated expression of multiple genes. However, cells from patient 16 showed increased expression in several regions. Going back to the paper, the researchers hypothesize that cells release chaperone systems to combat the protein aggregation that results from the knockout of ATM kinase (Huiting et al, 2021). The increased expression in particular regions could reflect the presence of these chaperone systems.


# Citations

Gu Z, Gu L, Eils R, Schlesner M, Brors B (2014). “circlize implements and enhances circular visualization in R.” Bioinformatics, 30, 2811-2812.

Gu Z, Eils R, Schlesner M (2016). “Complex heatmaps reveal patterns and correlations in multidimensional genomic data.” Bioinformatics.

Huber W, Carey VJ, Gentleman R, Anders S, Carlson M, Carvalho BS, Bravo HC, Davis S, Gatto L, Girke T, Gottardo R, Hahne F, Hansen KD, Irizarry RA, Lawrence M, Love MI, MacDonald J, Obenchain V, Ole's AK, Pag'es H, Reyes A, Shannon P, Smyth GK, Tenenbaum D, Waldron L, Morgan M (2015). “Orchestrating high-throughput genomic analysis with Bioconductor.” Nature Methods, 12(2), 115–121. http://www.nature.com/nmeth/journal/v12/n2/full/nmeth.3252.html.

Huiting, W., Dekker, S. L., Joris C. J. van der Lienden, Mergener, R., Furtado, G. V., Gerrits, E., Musskopf, M. K., Oghbaie, M., Stefano, L. H. D., Waarde-Verhagen, M. A. W. H. van, Couzijn, S., Barazzuol, L., LaCava, J., Kampinga, H. H., & Bergink, S. (2021, January 1). Targeting DNA topoisomerases or checkpoint kinases results in an overload of chaperone systems, triggering aggregation of a metastable subproteome. bioRxiv. Retrieved February 18, 2022, from https://www.biorxiv.org/content/10.1101/2021.06.04.447142v1

Isserlin, R. (2022). Differential Expression. Retrieved March 18, 2022, from  https://q.utoronto.ca/courses/248455/files/19309003?module_item_id=3480535

Isserlin, R. (2022). Annotation Dataset and Intro to Pathway analysis. Retrieved March 18, 2022, from  https://q.utoronto.ca/courses/248455/files/19306012?module_item_id=3480061

Kolberg L, Raudvere U, Kuzmin I, Vilo J, Peterson H (2020). “gprofiler2– an R package for gene list functional enrichment analysis and namespace conversion toolset g:Profiler.” F1000Research, 9 (ELIXIR)(709). R package version 0.2.1.

Li, M., & Liu, Y. (2016). Topoisomerase I in Human Disease Pathogenesis and Treatments. Genomics, proteomics & bioinformatics, 14(3), 166–171. https://doi.org/10.1016/j.gpb.2016.02.004 

Mitrea, D.M., Cika, J.A., Stanley, C.B. et al. (2018). Self-interaction of NPM1 modulates multiple mechanisms of liquid–liquid phase separation. Nat Commun 9, 842. https://doi.org/10.1038/s41467-018-03255-3 

Morgan, M. (2021). BiocManager: Access the Bioconductor Project Package Repository. R package version 1.30.16. https://CRAN.R-project.org/package=BiocManager

Ritchie ME, Phipson B, Wu D, Hu Y, Law CW, Shi W, Smyth GK (2015). “limma powers differential expression analyses for RNA-sequencing and microarray studies.” Nucleic Acids Research, 43(7), e47. doi: 10.1093/nar/gkv007.

Xie Y (2021). knitr: A General-Purpose Package for Dynamic Report Generation in R. R package version 1.37, https://yihui.org/knitr/.

